src/stores/store/secrets/hashicorp/hashicorp.go

```diff
@@ -135,23 +135,79 @@ func (s *Store) List(_ context.Context) ([]string, error) {
 }
 
 func (s *Store) Delete(_ context.Context, id string) error {
-	return errors.ErrNotImplemented
+	logger := s.logger.With("id", id)
+
+	var callData map[string][]string
+	hashicorpSecretData, err := s.client.Read(s.pathData(id), callData)
+	if err != nil {
+		errMessage := "failed to get Hashicorp secret data for deletion"
+		logger.WithError(err).Error(errMessage)
+		return errors.FromError(err).SetMessage(errMessage)
+	}
+	if hashicorpSecretData == nil {
+		errMessage := "Hashicorp secret not found for deletion"
+		logger.WithError(err).Error(errMessage)
+		return errors.NotFoundError(errMessage)
+	}
+
+	err = s.client.Delete(s.pathData(id))
+	if err != nil {
+		errMessage := "failed to delete Hashicorp secret"
+		logger.WithError(err).Error(errMessage)
+		return errors.FromError(err).SetMessage(errMessage)
+	}
+
+	return nil
 }
 
 func (s *Store) GetDeleted(_ context.Context, id string) (*entities.Secret, error) {
 	return nil, errors.ErrNotImplemented
 }
 
-func (s *Store) ListDeleted(ctx context.Context) ([]string, error) {
+func (s *Store) ListDeleted(_ context.Context) ([]string, error) {
 	return nil, errors.ErrNotImplemented
 }
 
-func (s *Store) Undelete(ctx context.Context, id string) error {
-	return errors.ErrNotImplemented
+func (s *Store) Undelete(_ context.Context, id string) error {
+	logger := s.logger.With("id", id)
+
+	// @TODO Fetch versions from DB once they are available
```

Comment:
version should be passed as argument in the secrets interfaces (like the Get endpoint).

Reply:
As discussed versions were removing in the past believing key vaults only supported "latest", since hashicorp required it I will be re-adding them on every secret store interface keeping empty as value of "latest"